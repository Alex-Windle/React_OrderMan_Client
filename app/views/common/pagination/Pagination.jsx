/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 *
 * @description The Pagination component represents a reusable pagination component that can be used with tables and lists. It can use Bootstrap's pagination component, with a variable number of pages, and click handlers to change the page.
 *
 * PropTypes = {
 *   handlePageChange: PropTypes.func.isRequired,
 *   pagination: PropTypes.shape({
 *     total: PropTypes.number,
 *     limit: PropTypes.number,
 *     page: PropTypes.number,
 *     pages: PropTypes.number,
 *   }).isRequired,
 * };
 *
 * DefaultProps = {};
 *
 * @exports The Pagination component
 */
