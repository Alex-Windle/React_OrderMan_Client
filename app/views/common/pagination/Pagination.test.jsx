/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React from 'react';
import renderer from 'react-test-renderer';


// Component imports
import Pagination from './Pagination';


// Data imports
import {
  testPagination,
} from '../../../spec/specData';


describe('Pagination component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Pagination
        handlePageChange={jest.fn()}
        pagination={testPagination}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
