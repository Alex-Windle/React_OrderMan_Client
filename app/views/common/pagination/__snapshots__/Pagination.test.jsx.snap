// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      className="pagination"
    >
      <li
        className="disabled"
      >
        <a
          onClick={[Function]}
          role="button"
        >
          «
        </a>
      </li>
      <li
        className="active"
      >
        <a
          onClick={[Function]}
          role="button"
        >
          1
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          role="button"
        >
          2
        </a>
      </li>
      <li
        className=""
      >
        <a
          disabled={false}
          onClick={[Function]}
          role="button"
        >
          »
        </a>
      </li>
    </ul>
  </div>
  <div
    className="container-fluid"
  >
    Showing Results 
    1
     - 
    10
  </div>
</div>
`;
